### PR TITLE
[5.2] Fix logout when user doesn't use the "remember me" functionality

### PR DESCRIPTION
**Pull request by KennedyTedesco**
Fixes: https://github.com/laravel/framework/issues/12054

### CODE REVIEW

File: src/Illuminate/Auth/SessionGuard.php [hunk 1]
```diff
@@ -561,7 +561,11 @@ public function logout()
         // listening for anytime a user signs out of this application manually.
         $this->clearUserDataFromStorage();
 
-        if (! is_null($this->user)) {
+        $ableToRefreshToken = ! is_null($this->user) && array_key_exists(
+            $user->getRememberTokenName(), $this->user->getAttributes()
```
Comment (crynobone) [inline src/Illuminate/Auth/SessionGuard.php]: Does this work with GenericUser provider?

Comment (KennedyTedesco) [inline src/Illuminate/Auth/SessionGuard.php]: Partially: https://github.com/laravel/framework/blob/5.2/src/Illuminate/Auth/GenericUser.php#L5

The problem is with `getAttributes()` that not necessarily exists on a GenericUser class, right? In that case I have to make a little change.

**Comment by KennedyTedesco** [comment]
@crynobone  I guess this is the best we can do right now. Any additional thoughts?

**Comment by taylorotwell** [comment]
So if the remember token is null it will never be refreshed? How would it ever be populated in the first place?

**Comment by taylorotwell** [comment]
Just put a remember_token on your database table if you are autenticating that model.

**Comment by KennedyTedesco** [comment]
In fact, it would be populated at login (if requested). But that's fair! To me isn't a problem. 👍  But I have saw a lot of people on forums with this issue (that generates a Query Exception).

**Pull request closed without merging**